sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/ScalarFunctionImpl.java

```diff
@@ -139,7 +139,7 @@ public CallImplementor getImplementor() {
 
     private static List<Expression> translate(List<Type> types, List<Expression> expressions) {
       Preconditions.checkArgument(
-          types.size() == expressions.size(), "types.size() != expressions.size()");
+          types.size() >= expressions.size(), "types.size() < expressions.size()");
```

Comment:
Could you explain a bit on this change? I am not quite following how this will fix those tests?

Reply:
Also, can we move this check completely?

Reply:
Concat function with only 5 arguments is registered: "org/apache/beam/sdk/extensions/sql/zetasql/BeamBuiltinMethods.java", but there are other concat functions that handle various number of arguments (1, 2, 3, 4, 5): "org/apache/beam/sdk/extensions/sql/zetasql/StringFunctions.java".

When using concat with less arguments (than it was registered with), Calcite will generate a call to the method with the correct number of arguments, but the precondition check will fail, since it compares the number of arguments passed to the number of arguments registered function accepts (always 5 for concat).

Reply:
I believe the right resolution is to register a var-arg version of function to Calcite. For example, for concat, the expected input should be:
types: [String[].class]
expression: [param1, param2, ...]
We will still be able to verify the input types and expression.

However my recall is Calcite doesn't support var-arg in user defined function (maybe I am wrong). So if we cannot define the var-arg version, we actually should remove this check as it's not longer useful.

Reply:
This check is still useful, it is protecting the for loop right after it. It is just too restrictive.

Reply:
The following loop was written based on the assumption that types = expression so each expression has a corresponding type. The check was based on the same assumption.

But now the assumption is proven to be wrong.

For example, I can see a valid input is:
types: [String[].class]
expression: [param1, param2, ...]

so the for loop will need to be modified to make this input work(every param match with String type). The loop happens to be right with `types.size() >= expressions.size()`.

I don't want to make this PR a one that needs long time to fix something. So if we want to check-in `types.size() >= expressions.size()`, at least can @11moon11 add comments to document what is going on, why need this change and leave a TODO in the for loop with a JIRA to support user defined function with var-arg

Reply:
I did add comments, please take a look when you have a chance.

Reply:
LGTM thanks.